spec/requests/api/requests/user_can_mark_a_request_complete_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+RSpec.describe 'Api::MyRequest::Requests :update', type: :request do
+  let(:user) { create(:user, email: 'user@email.com') }
+  let(:user2) { create(:user, email: 'user2@email.com') }
+  let(:credentials) { user.create_new_auth_token }
+  let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(credentials) }
+  let(:user2_credentials) { user2.create_new_auth_token }
+  let(:user2_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(user2_credentials) }
+  let(:request) { create(:request, requester: user) }
+
+  describe 'User can mark request as completed ' do
+    before do
+      put "/api/my_request/requests/#{request.id}", headers: headers, params: { activity: 'complete' }
+      request.reload
+    end
+
+    it 'has 200 response' do
+      expect(response).to have_http_status 200
+    end
+
+    it 'changes the status of the request' do
+      expect(request.status).to eq 'completed'
+    end
+
+    it 'responds with the completed confirmation' do
+      expect(response_json['message']).to eq 'Request completed!'
+    end
+  end
+
+  describe "User can't mark another user's request as completed" do
```

Comment:
"User can't mark a pending request as completed" would probably be a good sad path test

Reply:
Yes, Marcus's suggestion for a sad path is very relevant.

Reply:
done